src/main/java/io/vertx/core/impl/AddressResolver.java

```diff
@@ -131,14 +133,18 @@ public void close() {
   public void resolveHostname(String hostname, Handler<AsyncResult<InetAddress>> resultHandler) {
     ContextInternal callback = (ContextInternal) vertx.getOrCreateContext();
     io.netty.resolver.AddressResolver<InetSocketAddress> resolver = resolverGroup.getResolver(callback.nettyEventLoop());
-    io.netty.util.concurrent.Future<InetSocketAddress> fut = resolver.resolve(InetSocketAddress.createUnresolved(hostname, 80));
+    io.netty.util.concurrent.Future<InetSocketAddress> fut = resolver.resolve(InetSocketAddress.createUnresolved(hostname, 0));
     fut.addListener(a -> {
       callback.runOnContext(v -> {
         if (a.isSuccess()) {
           InetSocketAddress address = fut.getNow();
           resultHandler.handle(Future.succeededFuture(address.getAddress()));
         } else {
-          resultHandler.handle(Future.failedFuture(a.cause()));
+          if (Utils.isWindows() && hostname.equals("localhost")) {
```

Comment:
in the more recent version there is not comparison anymore, what is done instead is to add "alias=127.0.0.1" in the "hosts" Map, so I think this is handled by the underlying resolver, can you check ?